aszczachor/src/components/navigation/MobileMenu.js

```diff
@@ -0,0 +1,72 @@
+import styles from "components/navigation/mobileMenu.module.css";
+import { Link } from "react-router-dom";
+import { useState, useEffect, useRef } from "react";
+import ExpandButton from "./ExpandButton";
+
+const MobileMenu = ({
+  fashionStyling,
+  collections,
+  links,
+  openMenu,
+  menuBtn,
+}) => {
+  const [openFashionStyling, setOpenFashionStyling] = useState(false);
+  const [openCollections, setOpenCollections] = useState(false);
+  const ref = useRef();
+
+  useEffect(() => {
+    const onOutsideClick = ({ target }) => {
+      if (
+        (ref.current && ref.current.contains(target)) ||
+        (menuBtn.current && menuBtn.current.contains(target))
+      ) {
+        return;
+      }
+      openMenu(false);
+    };
+    document.addEventListener("click", onOutsideClick, true);
+    return () => document.removeEventListener("click", onOutsideClick, true);
```

Comment:
to akurat good job, jak już dodajemy listenery trzeba pamiętać, że w przypadku unmout trzeba je usunąć, gdybyś tego nie zrobiłby miałbys memory leak, poczytaj o tym